src/components/CurrentWeather.vue

```diff
@@ -30,35 +42,79 @@
 <script lang="ts">
 import { computed, defineComponent, ref } from 'vue';
 import { useWeatherStore } from '../stores/weather';
+import { useCitiesStore } from "../stores/cities";
 
 export default defineComponent({
 	setup() {
-		const store = useWeatherStore();
+		const weatherStore = useWeatherStore();
+		const citiesStore = useCitiesStore();
 		const city = ref('');
 		const hasSearched = ref(false);
+		const showSuggestions = ref(true);
 
-		const getWeather = async (): Promise<void> => {
+		const getWeather = async () => {
 			hasSearched.value = true;
-			store.setLoading(true);
+			weatherStore.setLoading(true);
 			try {
-				const response: any = await store.fetchWeather(city.value);
-				store.setWeather(response.data);
+				const response: any = await weatherStore.fetchWeather(city.value);
+				weatherStore.setWeather(response.data);
 			} catch (error: any) {
-				store.setError(error.message);
+				city.value = '';
 			} finally {
-				store.setLoading(false);
+				weatherStore.setLoading(false);
 			}
 		};
 
+		const fetchCities = async () => {
+			showSuggestions.value = true;
+			if (city.value.length >= 3) {
+				try {
+					const citiesArray: any[] = await citiesStore.fetchCities(city.value);
+					if (citiesArray) {
+						citiesStore.setCities(citiesArray);
+					} else {
+						console.error('No response received from fetchCities');
+					}
+				} catch (error) {
+					console.error(error);
+					citiesStore.setCities([]);
+				}
+			}
+		};
+
+		const selectCity = (selectedCity: any) => {
+			city.value = selectedCity.name;
+			citiesStore.setCities([]);
+			getWeather();
+		};
+
+		const handleEnterKey = () => {
+			getWeather();
+			hideSuggestions();
+		}
+
+		const hideSuggestions = () => {
+			showSuggestions.value = false;
+		}
+
 		return {
 			city,
 			getWeather,
-			weather: computed(() => store.weatherData),
-			error: computed(() => store.weatherError),
-			isLoading: computed(() => store.isLoading),
+			selectCity,
+			fetchCities,
+			handleEnterKey,
+			weather: computed(() => weatherStore.weatherData),
+			error: computed(() => weatherStore.weatherError),
+			isLoading: computed(() => weatherStore.isLoading),
 			hasSearched,
+			showSuggestions,
+			hideSuggestions,
+			cities: computed(() => citiesStore.getCities),
 		};
 	},
+	beforeMount() {
```

Comment:
it's a good practive to remove all event listeners on equivalent onUnmounted lifecycle hook